src/Microsoft.Azure.WebJobs.Logging/LogFactory.cs

```diff
@@ -15,23 +13,35 @@ public static class LogFactory
         /// <summary>
         /// Get a reader that reads from the given table. 
         /// </summary>
-        /// <param name="table"></param>
+        /// <param name="tableLookup"></param>
         /// <returns></returns>
-        public static ILogReader NewReader(CloudTable table)
+        public static ILogReader NewReader(ILogTableProvider tableLookup)
         {
-            return new LogReader(table);
+            return new LogReader(tableLookup);
         }
 
         /// <summary>
         /// Create a new writer for the given compute container name that writes to the given table.
         /// Multiple compute instances can write to the same table simultaneously without interference. 
         /// </summary>
         /// <param name="computerContainerName">name of the compute container. Likley %COMPUTERNAME%. </param>
-        /// <param name="table">underlying azure storage table to write to.</param>
+        /// <param name="tableLookup">underlying azure storage table to write to. 
+        /// Passed a string suffix (which will consist of valid Azure table characters). 
+        /// This must be a deterministic (replayable) function.</param>
         /// <returns></returns>
-        public static ILogWriter NewWriter(string computerContainerName, CloudTable table)
+        public static ILogWriter NewWriter(string computerContainerName, ILogTableProvider tableLookup)
```

Comment:
I'll switch the other places to computerContainerName. "ContainerName" is too overloaded and easily confused with Azure Storage. 

---
In reply to: [82909392](https://github.com/Azure/azure-webjobs-sdk/pull/862#discussion_r82909392) [](ancestors = 82909392)